### ISSUE TITLE
orthogonalComplement in complexEigs.js should return a vector but returns a matrix

### ISSUE THREAD

**Issue by vjancik** [josdejong/mathjs#2927]
**Describe the bug**
I'm using `math.eigs` by passing a 2D row-major matrix array.
```javascript
function arrMatToRows(arrMat, res = [new Array(4), new Array(4), new Array(4), new Array(4)]) {
  const dim = Math.sqrt(arrMat.length);
  for (let i = 0; i < dim; ++i) {
    for (let j = 0; j < dim; ++j) {
      res[i][j] = arrMat[j*dim + i];
    }
  }
  return res;
}

eigs = math.eigs(arrMatToRows(mZoomInvert, rowBuff));
```
Line numbers are from `https://cdnjs.cloudflare.com/ajax/libs/mathjs/11.7.0/math.js`
It throws with 
```
Uncaught RangeError: Expected a column vector, instead got a matrix of size (4, 4)
    o dot.js:46
    Array | DenseMatrix, Array | DenseMatrix dot.js:64
    me typed-function.js:1794
    C complexEigs.js:681
    M complexEigs.js:696
    A complexEigs.js:660
    E complexEigs.js:618
    I complexEigs.js:448
    M complexEigs.js:439
    z complexEigs.js:52
    q eigs.js:100
    Array eigs.js:51
    me typed-function.js:1794
    <anonymous> pen.js:258
    EventListener.handleEvent* pen.js:235
dot.js:46:12
```

I've traced the problem to `orthogonalComplement` at `complexEigs.js:672`.
Here is argument v at the beginning of the function call:
![bugBeforeMathjs](https://user-images.githubusercontent.com/4226865/228863086-368e5573-4715-48db-8546-85e6324fe378.png)
and here it is at the end (which ends up throwing in the dot function from the norm call), including the ortho argument :
![bugAfterMathjs](https://user-images.githubusercontent.com/4226865/228863596-2a9bdee5-5d71-4fdb-8e2f-a09c8ef97eed.png)

**To Reproduce**
The error is thrown [in this demo app](https://codepen.io/vjancik/full/vYzKZrq) but it requires a specific combination of zoom + pan + change rotation origin + rotate while un-zooming to occur. It appears the function only gets invoked when round-off errors accumulate and distort the transformation matrix.
**EDIT:** I've since changed the code, possibly removing use of `math.eigs`, you can copy paste the code (from the bug occurrence) from [here](https://gist.github.com/vjancik/98faf183e332d1f793fd997e150eb017). You also have to add the previously quoted `math.js` script in *Settings > JS*

**Sidenote**
I've also tried to call `math.eigs` with a matrix returned from 
```
const temp = math.matrix(mZoomInvert);
temp.resize([4, 4]);
eigs = math.eigs(temp);
```
but that threw every time. Is that also worth exploring or is that normal?

**Comment by josdejong** [comment]
Thanks for reporting Victor. Can you create a minimal example demonstrating the issue? Something that can be tried out straight in the developer console of https://mathjs.org/ for example, like:

```js
const matrix = [...] // array or nested array containing numbers
const eigs = math.eigs(matrix)
```

In the examples you post, there are arrays like `mZoomInvert` and `rowBuff` from which I do not know what they contain and what type they are.

**Comment by gwhitney** [comment]
This will be fixed by the fix to #2879.

**Comment by josdejong** [comment]
Fixed in `v12.0.0` now via #3037
